apps/hash-graph/lib/graph/src/api/rest/data_type.rs

```diff
@@ -200,6 +172,58 @@ where
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+struct LoadExternalDataTypeRequest {
+    #[schema(value_type = String)]
+    data_type_id: VersionedUrl,
+    actor_id: RecordCreatedById,
+}
+
+#[utoipa::path(
+    post,
+    path = "/data-types/load",
+    request_body = LoadExternalDataTypeRequest,
+    tag = "DataType",
+    responses(
+        (status = 200, content_type = "application/json", description = "The metadata of the loaded data type", body = OntologyElementMetadata),
+        (status = 422, content_type = "text/plain", description = "Provided request body is invalid"),
+
+        (status = 409, description = "Unable to load data type in the store as the base data type ID already exists"),
+        (status = 500, description = "Store error occurred"),
+    ),
+)]
+#[tracing::instrument(level = "info", skip(pool, domain_validator))]
+async fn load_external_data_type<P: StorePool + Send>(
+    pool: Extension<Arc<P>>,
+    domain_validator: Extension<DomainValidator>,
+    body: Json<LoadExternalDataTypeRequest>,
+) -> Result<Json<OntologyElementMetadata>, StatusCode>
+where
+    for<'pool> P::Store<'pool>: RestApiStore,
+{
+    let mut store = pool.acquire().await.map_err(|report| {
+        tracing::error!(error=?report, "Could not acquire store");
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?;
+
+    #[allow(clippy::infallible_destructuring_match)]
```

Comment:
I've seen this before, do you know why it's complaining? I don't see a `match`, also do we perhaps want to use `expect` instead?

Reply:
Previously, with the type fetcher being a compile-time option, the second arm of the match was was compiled out. As this lint only happened without the type fetcher enabled I couldn't use `expect` because otherwise it would complain if the type fetcher is enabled.

That I still had it here was a copy-paste error, but as the type fetcher is not compiled conditionally it's not neither regardless. Thanks for spotting! [`f58d430` (#2600)](https://github.com/hashintel/hash/pull/2600/commits/f58d43023bc81dffb621dd39b8e2c0fd98169f4e)